esmvalcore/_recipe.py

```diff
@@ -1275,6 +1376,11 @@ def initialize_tasks(self):
                     config_user=self._cfg,
                     task_name=task_name,
                 )
+                # remove possible duplicate ancestor tasks that
+                # preprocess the same
+                # fx var file needed by different var["activity"] but with
+                # the same output fx var file
+                _check_duplication(task)
```

Comment:
Are we sure that this doesn't remove useful error messages?

Reply:
It does cloak the RecipeError for duplicate preprocessor files, yes, but then again it also removes the duplication itself. I reckon this way it's way more useful. We could add an extra layer of check (if fx in filename) but I dont think it's needed